rig-neo4j/examples/vector_search_simple.rs

```diff
@@ -0,0 +1,138 @@
+    providers::openai::{Client, TEXT_EMBEDDING_ADA_002},
+    vector_store::VectorStoreIndex as _,
+};
+use rig_neo4j::{
+    vector_index::{IndexConfig, SearchParams},
+    Neo4jClient, ToBoltType,
+};
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    // Initialize OpenAI client
+    let openai_api_key = env::var("OPENAI_API_KEY").expect("OPENAI_API_KEY not set");
+    let openai_client = Client::new(&openai_api_key);
+
+    // Initialize Neo4j client
+    let neo4j_uri = env::var("NEO4J_URI").expect("NEO4J_URI not set");
+    let neo4j_username = env::var("NEO4J_USERNAME").expect("NEO4J_USERNAME not set");
+    let neo4j_password = env::var("NEO4J_PASSWORD").expect("NEO4J_PASSWORD not set");
+
+    let neo4j_client = Neo4jClient::connect(&neo4j_uri, &neo4j_username, &neo4j_password).await?;
+
+    // Select the embedding model and generate our embeddings
+    let model = openai_client.embedding_model(TEXT_EMBEDDING_ADA_002);
+
+    let embeddings = EmbeddingsBuilder::new(model.clone())
+        .simple_document("doc0", "Definition of a *flurbo*: A flurbo is a green alien that lives on cold planets")
+        .simple_document("doc1", "Definition of a *glarb-glarb*: A glarb-glarb is a ancient tool used by the ancestors of the inhabitants of planet Jiro to farm the land.")
+        .simple_document("doc2", "Definition of a *linglingdong*: A term used by inhabitants of the far side of the moon to describe humans.")
+        .build()
+        .await?;
+
+    // The struct that will reprensent a node in the database. Used to deserialize the results of the query (passed to the `top_n` methods)
+    // ❗IMPORTANT: The field names must match the property names in the database
+    #[derive(serde::Deserialize)]
+    struct Document {
+        #[allow(dead_code)]
+        id: String,
+        document: String,
+        #[allow(dead_code)]
+        embedding: Vec<f32>,
+    }
+
+    let _create_nodes = futures::future::join_all(embeddings.into_iter().map(|doc| {
+        neo4j_client.graph.run(
+            neo4rs::query(
+                "
+                    CREATE
+                        (document:DocumentEmbeddings {
+                            id: $id,
+                            document: $document,
+                            embedding: $embedding})
+                    RETURN document",
+            )
+            .param("id", doc.id)
+            // Here we use the first embedding but we could use any of them.
+            // Neo4j only takes primitive types or arrays as properties.
+            .param("embedding", doc.embeddings[0].vec.clone())
+            .param("document", doc.document.to_bolt_type()),
+        )
+    }))
+    .await;
+
+    // Unwrap the results in the vector _create_nodes
+    for result in _create_nodes {
+        result.unwrap(); // or handle the error appropriately
+    }
+
+    // Create a vector index on our vector store
+    println!("Creating vector index...");
+    let _ = neo4j_client
```

Comment:
No need for `let _ = `

Reply:
true! I was mistakenly using .execute() instead of run() before.